File: FoodDelivery.Web/src/app/models/dish/dish.ts
import { Guid } from 'guid-typescript';
import { restaurant } from 'src/app/models/restaurant/restaurant';
import { dishCategory } from './dishCategory';

export class dish{
    public id?: Guid;
    public name: string;
    public description: string;
    public price: number;
    public rating?: number
    public weight: number;

    public restaurantId: Guid;
    public category: dishCategory;
}

File: FoodDelivery.DAL/EF/Entities/Dish.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace FoodDelivery.DAL.EF.Entities
{
	public class Dish
	{
		[Key]
		public Guid Id { get; set; }
		public string Name { get; set; }
		public string Description { get; set; }
		public decimal Price { get; set; }
		public double Weight { get; set; }

		public int Status { get; set; }

		[ForeignKey("Category")]
		public int CategoryId { get; set; }


		[ForeignKey("Restaurant")]
		public Guid RestaurantId { get; set; }

		public virtual DishCategory Category { get; set; }

		public virtual Restaurant Restaurant { get; set; }
		public virtual List<DishRating> Ratings { get; set; }
	}
}
